pages/api/system/reload.js

```diff
@@ -6,28 +6,33 @@ import connectMongo from "@config/mongo";
 import logger from "@config/logger";
 import Profile from "@models/Profile";
 import Link from "@models/Link";
+import { extendedProfileSchema } from "@config/schemas/validationSchemas";
 
 export default async function handler(req, res) {
-  if (req.method !== "GET" || req.query.secret !== serverEnv.LINKFREE_API_SECRET) {
+  if (
+    req.method !== "GET" ||
+    req.query.secret !== serverEnv.LINKFREE_API_SECRET
+  ) {
     logger.error(
       `attempt to load profile json but security check failed: "${req.query.secret}"`
     );
     return res.status(401).json({ error: "ONLY system calls allowed" });
   }
+
   await connectMongo();
 
   // 1. load all profiles
-  const basicProfiles = findAllBasic();
-  const fullProfiles = basicProfiles.map((profile) =>
-    findOneByUsernameFull(profile)
-  );
+  const basicProfiles = await findAllBasic();
+  const fullProfiles = await Promise.all(
+    basicProfiles.map((profile) => findOneByUsernameFull(profile))
+  ).then((profiles) => profiles.filter((profile) => profile !== null));
```

Comment:
Can this be changed to `await`

Reply:
I mean instead of using `then` to use `await`, this way it is consistent with the rest of the application